gapic-generator/lib/gapic/presenters/service_rest_presenter.rb

```diff
@@ -326,20 +326,42 @@ def is_main_mixin_service?
         main_service.is_main_mixin_service?
       end
 
+      ##
+      # Whether there are mixin services that should be referenced
+      # in the client for this service
+      #
+      # @return [Boolean]
+      #
+      def mixins?
+        main_service.mixins?
+      end
+
+      ##
+      # Whether there are mixin services that this package has http binding overrides for.
+      #
+      # @return [Boolean]
+      #
+      def mixin_binding_overrides?
+        main_service.mixin_binding_overrides?
+      end
+
       ##
       # The client presenters of the mixin services that are used by the methods of this service
       #
-      # @return [Enumerable<Gapic::Presenters::Service::LroClientPresenter>]
+      # @return [Enumerable<Gapic::Presenters::Service::MixinClientPresenter>]
       def mixin_presenters
         return [] unless main_service.mixins?
-        main_service.mixins.map do |mixin|
-          Gapic::Presenters::Service::LroClientPresenter.new service: mixin.service,
-                                                             client_class_name: mixin.client_class_name_rest,
-                                                             client_class_docname: mixin.client_class_docname_rest,
-                                                             client_var_name: mixin.client_var_name,
-                                                             require_str: mixin.require_str_rest,
-                                                             service_description: mixin.service_description,
-                                                             helper_type: ""
+        main_service.mixin_presenters.map do |grpc_presenter|
+          model = main_service.mixin_models.first { |mdl| mdl.service == grpc_presenter.service }
```

Comment:
`first` doesn't take a block. This will just return the first element in the list and ignore the block. You probably want `find` instead.

Reply:
:+1: